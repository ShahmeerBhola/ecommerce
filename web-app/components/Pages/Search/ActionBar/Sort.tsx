import React, { FC } from 'react';
import { useRouter } from 'next/router';

import { getQueryParams, getPagePath } from '../../../../utils';

import { FancySelect } from '../../../Common';

const sortOptions: Forms.FieldOption<WordPress.ProductSortOption>[] = [
    {
        label: 'Price (Low to High)',
        value: 'price-lth',
    },
    {
        label: 'Price (High to Low)',
        value: 'price-htl',
    },
    {
        label: 'Popular',
        value: 'popular',
    },
    {
        label: 'Newest',
        value: 'newest',
    },
];

export const Sort: FC = () => {
    const router = useRouter();

    const queryParams = getQueryParams(router);

    const sortHandler = (value: string): void => {
        if (value) {
            if (queryParams.sort !== value) {
                router.push({ pathname: getPagePath(), query: { ...queryParams, sort: value } });
            }
        } else if ('sort' in queryParams) {
            // eslint-disable-next-line @typescript-eslint/no-unused-vars
            const { sort: __, ...newQueryParams } = queryParams;
            router.push({ pathname: getPagePath(), query: newQueryParams });
        }
    };

    return (
        <div>
            <FancySelect
                options={sortOptions}
                label="Sort by"
                defaultValue="default"
                isPlain
                onSelect={sortHandler}
                currentValue={queryParams.sort || null}
            />
        </div>
    );
};
