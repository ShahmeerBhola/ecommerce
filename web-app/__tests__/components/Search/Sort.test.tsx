import React from 'react';
import renderer from 'react-test-renderer';
import Sort from '../../../components/Search/Sort';

describe('Sort component', () => {
  const onChange = (): void => console.log('hello');

  test('it renders properly when value is blank string', () => {
    const tree = renderer.create(<Sort value="" onChange={onChange} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when value is price-lth', () => {
    const tree = renderer.create(<Sort value="price-lth" onChange={onChange} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
