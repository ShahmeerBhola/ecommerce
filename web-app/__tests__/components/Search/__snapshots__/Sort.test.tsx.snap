// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort component it renders properly when value is blank string 1`] = `
<select
  className="text-sm text-gray-200 p-3 focus:outline-none w-full h-50px bg-gray-50 border border-solid border-gray-100"
  name="sort"
  onChange={[Function]}
  value="Sort"
>
  <option
    disabled={true}
    value="Sort"
  >
    Sort
  </option>
  <option
    value="price-lth"
  >
    Price (Low to High)
  </option>
  <option
    value="price-htl"
  >
    Price (High to Low)
  </option>
  <option
    value="popular"
  >
    Popular
  </option>
  <option
    value="newest"
  >
    Newest
  </option>
</select>
`;

exports[`Sort component it renders properly when value is price-lth 1`] = `
<select
  className="text-sm text-gray-200 p-3 focus:outline-none w-full h-50px bg-gray-50 border border-solid border-gray-100"
  name="sort"
  onChange={[Function]}
  value="price-lth"
>
  <option
    disabled={true}
    value="Sort"
  >
    Sort
  </option>
  <option
    value="price-lth"
  >
    Price (Low to High)
  </option>
  <option
    value="price-htl"
  >
    Price (High to Low)
  </option>
  <option
    value="popular"
  >
    Popular
  </option>
  <option
    value="newest"
  >
    Newest
  </option>
</select>
`;
